#!/usr/bin/env python
PACKAGE = "hlpr_lookat"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("h_min", int_t, 0, "Hue minimum value", 0, 0, 255)
gen.add("h_max", int_t, 0, "Hue maximum value", 255, 0, 255)
gen.add("s_min", int_t, 0, "Saturation minimum value", 0, 0, 255)
gen.add("s_max", int_t, 0, "Saturation maximum value", 255, 0, 255)
gen.add("v_min", int_t, 0, "Value minimum value", 0, 0, 255)
gen.add("v_max", int_t, 0, "Value maximum value", 255, 0, 255)
gen.add("opening_diam", int_t, 0, "Diameter of circle for denoising", 3, 1, 100)
gen.add("closing_diam", int_t, 0, "Diameter of circle for closing forms", 3, 1, 100)
gen.add("min_final_size", int_t, 0, "Minimum size of object after processing", 100, 1, 1000)

exit(gen.generate(PACKAGE, "hlpr_lookat", "ColorTrack"))